### PR TITLE
testing/logstash: update to 5.2.2 / use supervise-daemon

### PR DESCRIPTION
**Pull request by itoffshore**
adds service supervision with OpenRC's builtin `supervise-daemon`
`logstash/bin` added to `$PATH`
take ownership

### CODE REVIEW

**Comment by itoffshore** [comment]
`supervise-daemon` restarts `logstash` ok when the process is killed.

It's not clear [in the guide](https://github.com/OpenRC/openrc/blob/master/supervise-daemon-guide.md) but each service needs it's own pid - perhaps use a format of `$SVCNAME.sd.pid`

File: testing/logstash/APKBUILD [hunk 1]
```diff
@@ -1,64 +1,88 @@
...
+pkgver=5.2.1
+pkgrel=0
 pkgdesc="A flexible, open source, data collection, parsing and enrichment pipeline"
 url="http://logstash.net"
-arch="noarch"
+arch="all"
 license="apache"
-depends="openjdk7-jre-base"
-depends_dev=""
+depends="openjdk8-jre-base libc6-compat"
```
Comment (kaniini) [inline testing/logstash/APKBUILD]: why is libc6-compat needed?  is there now native JNI code?

same for `$arch`

File: testing/logstash/logstash.conf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+input {
+        file {
+        type => "syslog"
+        path => ["/var/log/**/*.log"]
+        }
+}
+
+output {
+        elasticsearch { hosts => ["localhost:9200"] }
```
Comment (kaniini) [inline testing/logstash/logstash.conf]: this does not seem like a reasonable default as the package does not have a hard dependency on ES

File: testing/logstash/APKBUILD (hunk 1 above)
Comment (itoffshore) [inline testing/logstash/APKBUILD]: without `libc6-compat` tailing files does not work

File: testing/logstash/logstash.conf (hunk 2 above)
Comment (itoffshore) [inline testing/logstash/logstash.conf]: I just used the settings from the [ES reference docs](https://www.elastic.co/guide/en/logstash/current/configuration.html) - have commented out `elasticsearch`.

I also changed the log directory in the `confd` so logstash does not consume it's own logs

By default the `stdout` plugin prints to the `/var/lib/logstash/logs/error.log`)

**Comment by jirutka** [comment]
If it contains JARs with native binaries, then these should be removed. Please read at least https://github.com/jruby/jruby/wiki/JRuby-on-Alpine-Linux

**Comment by jirutka** [comment]
BTW I gave up packaging Logstash for a reason... I hope that the new version is not so horrible ball of mud as it was a year ago, otherwise it'd be just a waste of time.

**Comment by itoffshore** [comment]
@jirutka - ok - I'll check which `jar` is built against `glibc` & build a native version

**Comment by jirutka** [comment]
Wait a moment, IIRC Logstash runs on JRuby and there's no jruby in dependencies. Aren't you bundling it, are you!?

**Comment by jirutka** [comment]
If they haven't pulled another deps with native extensions, then we already have all the needed java libs with native extensions in aports.

**Comment by itoffshore** [comment]
I was using the `vendor` libraries that come with `logstash` - I'll try & work out which one is the problem

**Comment by itoffshore** [comment]
it uses `jffi` which need `libc6-compat` which I have as a depends

**Comment by itoffshore** [comment]
I'll add your `java-jffi-native` as a depends & remove the bundled `jffi`

**Comment by jirutka** [comment]
Do not even consider bundling jruby or any JAR with included native binaries.

I've already made package for jffi:
https://pkgs.alpinelinux.org/package/edge/community/x86_64/java-jffi

Actually I did whole work with packaging JRuby just to make a package for Logstash. I haven't finished Logstash itself, b/c it was so messy, that it was nearly impossible to do it right.

File: testing/logstash/logstash.post-deinstall [hunk 3]
```diff
@@ -1,5 +1,6 @@
 #!/bin/sh
 
 deluser logstash 2>/dev/null
+rm -f /etc/profile.d/logstash 2>/dev/null
```
Comment (jirutka) [inline testing/logstash/logstash.post-deinstall]: What the... Maybe you want also remove whole home directory as well? Just for fun, users love when pkgs removes their files when uninstalling...

Comment (itoffshore) [inline testing/logstash/logstash.post-deinstall]: it's just removing `/usr/share/logstash/bin` from `$PATH`

Comment (itoffshore) [inline testing/logstash/logstash.post-deinstall]: I removed the `post-deinstall` script & fixed the merge conflict.

File: testing/logstash/logstash.conf (hunk 2 above)
Comment (itoffshore) [inline testing/logstash/logstash.conf]: `logstash.conf` has `elasticsearch` input commented out now  - can it be pulled ?

File: testing/logstash/APKBUILD (hunk 1 above)
Comment (itoffshore) [inline testing/logstash/APKBUILD]: fixed to use Alpine's `java-jffi-native` & `libc6-compat`

**Pull request closed without merging**